### ISSUE TITLE
Verify deposits, exits and slashings before queueing

### ISSUE THREAD

**Issue by paulhauner** [sigp/lighthouse#276]
## Description

Presently, whenever a `Deposit`, `Exit`, `AttesterSlashing` or `ProposerSlashing` is given to the `BeaconChain` it does not verify it before adding it to the "inclusion queue".

Therefore, if one submits an invalid object it will cause block production to fail indefinitely thereafter.

## Steps to resolve

Note: this code is currently in #264, do not start issue until it has been merged. Exits queues have not been implemented yet, they will come soon.

Add verification steps to the `BeaconChain::receive_x_for_inclusion` where `x = [deposit, attester_slashing, proposer_slashing, exit]` functions.

**Comment by paulhauner** [comment]
Closing in favor of #281
